### DISCUSSION TITLE
How to disable logs, even the ones from the error reporter?

### DISCUSSION THREAD

**Discussion by devfservant** [cypress-io/cypress#19109] [Questions and Help]
In my `cy.login` custom command implementation, I am doing something like this:

```ts
const payload = {
  url: tokenUrl,
  method: 'POST',
  body: {
    grant_type: 'password',
    client_secret: clientSecret,
    username,
    password,
    ...,
  },
};
return cy.request(payload);
```

It works well and I plan to put it in my pipeline using the `cypress run` command.  
My problem is that if the request fails, the Cypress error reporter will log the entire request body, even its secrets,  
which is of course problematic from a security point of view:

```
     CypressError: `cy.request()` failed on:
https://login.microsoftonline.com/myTenantId/oauth2/v2.0/token
The response we received from your web server was:
  > 401: Unauthorized
This was considered a failure because the status code was not `2xx` or `3xx`.
If you do not want status codes to cause failures pass the option: `failOnStatusCode: false`
The request we sent was:
Method: POST
URL: https://login.microsoftonline.com/myTenantId/oauth2/v2.0/token
Body: grant_type=password&client_id=myClientId&client_secret=myClientSecret&username=myUsername&password=myPassword&...
```

**How can I prevent Cypress from printing such detailed logs?**  
It tried:

```ts
return cy.request({ ...payload, log: false });
```

but it did not help.  
Finally, I ended up with:

```ts
const requestOptions = { ...payload, failOnStatusCode: false, log: false };
return cy.request(requestOptions).then((response) => {
  if (response.status === 200) {
    return response;
  } else {
    throw new Error(`Request ended with status ${response.status}`);
  }
});
```

but I am not satisfied with this solution.
I am afraid it may not be suitable for other cases of errors, which would log the entire request body.  
How would you proceed?

EDIT:

I saw the [`--quiet`](https://docs.cypress.io/guides/guides/command-line#cypress-run) flag which might help, but how would you do to silent _just_ the `cy.request` output?

**Comment by TatyanaVl** [comment]
Has anyone was able to come up with alternative solution?
It seems that there has to be a way to disable logs for the cy.request on a global level.
I have tried to use: `cy.intercept({ resourceType: /xhr|fetch/ }, { log: false }),` but that did not work.

**Reply by TatyanaVl** [reply]
It seems that the issue is with the reporter - see https://github.com/cypress-io/cypress/issues/20214

**Reply by devfservant** [reply]
I believe that this issue should be reopen

**Comment by devfservant** [comment]
FYI I've open https://github.com/cypress-io/cypress/issues/28406
